airflow/providers/amazon/aws/transfers/gcs_to_s3.py

```diff
@@ -117,7 +119,18 @@ def __init__(
     ) -> None:
         super().__init__(**kwargs)
 
-        self.bucket = bucket
+        if bucket:
+            warnings.warn(
+                "The 'bucket' parameter is deprecated and will be removed in a future version. "
+                "Please use 'gcs_bucket' instead.",
+                AirflowProviderDeprecationWarning,
+                stacklevel=2,
+            )
+            self.gcs_bucket = bucket
+        elif gcs_bucket:
```

Comment:
By just doing a if, if both are provided, we use `gcs_bucket`, which is what we want

```suggestion
        if gcs_bucket:
```

Reply:
Sorry, I missed that in your previous comment.